example/demo.py

```diff
@@ -27,26 +27,26 @@
 assert updatetextwindow is not None, 'Window was none'
 viz.text('And here it is', win=updatetextwindow, append=True)
 
-# audio demo:
-tensor = np.random.uniform(-1, 1, 441000)
-viz.audio(tensor=tensor, opts={'sample_frequency': 441000})
+# text window with Callbacks
+txt = 'This is a write demo notepad. Type below. Delete clears text:<br>'
+callback_text_window = viz.text(txt)
 
-# audio demo: download from http://www.externalharddrive.com/waves/animal/dolphin.wav
-try:
-    audio_url = 'http://www.externalharddrive.com/waves/animal/dolphin.wav'
-    # linux
-    if _platform == "linux" or _platform == "linux2":
-        audiofile = '/home/%s/dolphin.wav' % getpass.getuser()
-    # MAC OS X
-    elif _platform == "darwin":
-        audiofile = '/Users/%s/dolphin.wav' % getpass.getuser()
-    # download audio
-    urllib.request.urlretrieve(audio_url, audiofile)
 
-    if os.path.isfile(audiofile):
-        viz.audio(audiofile=audiofile)
-except BaseException:
-    print('Skipped audio example')
+def type_callback(event):
+    if event['eventType'] == 'keyPress':
+        curr_txt = event['paneData']['content']
+        if event['key'] == 'Enter':
+            curr_txt += '<br>'
+        elif event['key'] == 'Backspace':
+            curr_txt = curr_txt[:-1]
+        elif event['key'] == 'Delete':
+            curr_txt = txt
+        elif len(event['key']) == 1:
+            curr_txt += event['key']
+        viz.text(curr_txt, win=callback_text_window)
+
+
+viz.event_handlers[callback_text_window] = type_callback
```

Comment:
An alternative API would be to have a function `register_event_handler(handler, win)` that updates the `event_handlers` dict. The advantage of this would be that the function could check that the specified `win` exists (and raise an exception if it doesn't), that `iscallable(callback)`, and that it could compose callbacks of multiple callbacks are added to the same window.

Reply:
Yeah having it be an API call would be rather nice. I'm keeping away from function composition for now though b/c I don't think there's a good clean way to deal with decomposition. Could be dealt with later.

Reply:
A user may want to create a callback for a window that doesn't yet exist though, I feel this behavior is not degenerate as users are allowed to declare window names on creation.

Reply:
Wouldn't callback composition simply amount to running each of the callbacks sequentially in the order that they were added? Under the hood, `event_handlers[callback_text_window]` would maintain a list of callbacks rather than a single callback. `register_event_handler()` would simply append to the correct list (creating a new empty list if it doesn't exist yet).

Re callback for a window that doesn't yet exist though: I see, makes sense. In that case, the registering function would not have to check for `win` existence.

Reply:
Oh yeah that's exactly how it would work, my difficulty would be how to expose an API for _removing_ a callback that has already been added. What do you think would be a good approach on that front?

Reply:
Hmmm yeah that is tricky. Perhaps add a function `remove_callbacks(win)` that empties the table with callbacks for the specified `win` suffices? If users want more advanced behaviors, they can always implement the desired composition themselves (in the same way that they need to manually manipulate the content of `event_handlers` now)..?